### PR TITLE
Wrap, doc, and test git-cherrypick

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/repository.jl [hunk 1]
```diff
@@ -313,6 +313,24 @@ function checkout_head(repo::GitRepo; options::CheckoutOptions = CheckoutOptions
                  repo.ptr, Ref(options))
 end
 
+"""
+    LibGit2.cherrypick(repo::GitRepo, commit::GitCommit; options::CherrypickOptions = CherrypickOptions())
+
+Cherrypick the commit `commit` and apply the changes in it to the current state of `repo`.
+The keyword argument `options` sets checkout and merge options for the cherrypick. If `null`,
```
Comment (tkelman) [inline base/libgit2/repository.jl]: which `null` ?

Comment (kshyatt) [inline base/libgit2/repository.jl]: `C_NULL` but in fact we just use the default values anyway so I should probably delete that (it's leftover from when that kwarg was a `Nullable`, very nasty).

File: base/libgit2/repository.jl [hunk 2]
```diff
@@ -313,6 +313,24 @@ function checkout_head(repo::GitRepo; options::CheckoutOptions = CheckoutOptions
...
+the default options are used.
+
+!!! note
+    `cherrypick` will *apply* the changes in `commit` but not *commit* them, so `repo` will
+    be left in a dirty state. If you want to apply and commit the changes in `commit` you must
```
Comment (tkelman) [inline base/libgit2/repository.jl]: this cherrypick applies but does not commit, yeah?

Comment (kshyatt) [inline base/libgit2/repository.jl]: Yes, does the docstring not make this clear?

Comment (tkelman) [inline base/libgit2/repository.jl]: The first sentence does, but for the second sentence to sound consistent with it I'd say "If you want to also commit the changes"

Comment (kshyatt) [inline base/libgit2/repository.jl]: Cool, thanks and will fix :)

**Review by ararslan — APPROVED** [review]

**Pull request merged**